python/_infer.py

```diff
@@ -0,0 +1,249 @@
+from typing import Any
+from ._common import *
+from enum import IntEnum
+from collections.abc import Iterable, Union, Tuple, Mapping, Callable
+import abc
+import queue
+
+
+class Tensor:
```

Comment:
> Is it possible to have only a single Tensor class?

Can you elaborate on this? I think there will only be one `Tensor` class, to store the information needed for Triton (buffer attribute, tensor data etc.). Although the `InferenceRequest` accepts other tensor types that implements dlpack interface, internally it is converted to `Tensor` and then process with the Triton APIs. On this note I agree that we should have `Tensor.from_dlpack`. However, the conversion doesn't involve data copy so the numpy / other dlpack-compatible tensor must be valid.

The part about allocator in `Tensor` is related to the output buffer allocation. The user controls how the tensor data is allocated, that is probably where having multiple Tensor classes seems to be happening: even the `Tensor` interface is unchanged, each implementation of the allocator will have to set additional attributes in `Tensor` for allocation purpose.

Reply:
I see what you are saying now, the `NumpyAllocator` does extend `Tensor` class to add allocator specific detail instead of directly adding attributes to a `Tensor` instance. I think this is just a different way to achieve my earlier comment about buffer allocation, users should process the output buffer via `DLPack` / `Tensor` interface which is allocator agnostic, my example usage is not appropriate as it assumes the allocator detail..

Reply:
Exactly, I'm just saying that we should try to decouple the buffer allocation logic from the tensor representation.